rnnalyse/extractors/base_extractor.py

```diff
@@ -110,19 +111,25 @@ def extract(self,
                         all_activations[name].append(sen_activations[name])
 
                 extracted_labels.extend(sen_extracted_labels)
-                num_extracted += len(sen_extracted_labels)
+                activation_ranges[sen_id] = (n_extracted, n_extracted+len(sen_extracted_labels))
```

Comment:
So `activation_ranges` is  a dict mapping from sentence_id to the positions that were extracted? I suppose this comes in handy somewhere when using the new indexing for `ActivationsReader`?

Reply:
Yes indeed, this makes it possible to retrieve the activations of a specific sentence. Because of the selection_func (which is a fantastic addition btw!), the ranges can no longer be based on the sentence lengths.

I can imagine that in the future we want to write more relevant info to file besides these ranges, but for now I could only come up with the ranges being useful. Maybe dumping the actual extracted indices of a sentence is nice as well of course, because right now you only know which activations belong to some sentence, but not to the individual tokens themselves. I'll add it to the todo's :p

Reply:
I see, then this addition makes a lot of sense! Thanks!